ios/Classes/Common/EncoderError.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+
+internal enum EncoderError: Error {
+    indirect case log(EncoderError, String = #filePath, Int = #line)
+    case notImplemented
+    case notExist
+    case notArray
+    case notDictionary
+    case outOfRange
+    case decoderFailed
+    case encoderFailed
+    case storageFailed
+    case castingToStringFailed
+    case castingToNumberFailed
+    case createObjectFailed
```

Comment:
It may be useful to add a value, file path and line information to some of the cases. What do you think?

Reply:
they already have `file` and `path` by using `.log()`
Is there any particular reason to add `value`?

Reply:
I thought for a moment that you are using `.log` as means of storing underlying errors. But this is not the case. 

I find `log` as an error / exception type to be rather confusing. I think it will be easier to read this code if each of the cases had `let file, let lineNumber` as arguments.

As for the for the value, for the cases that represent an error during conversion for example, they could store the value that could not be converted, in order for this value to be included in the localised string or reported otherwise.